Allow blocking structured facts

djuarezg opened the pull request:
This patch works as well for structured facts, not like https://github.com/puppetlabs/puppetdb/pull/3998

This changeset adds two configuration options used by the facts PuppetDB indirector:

    fact_names_blocklist
    fact_names_blocklist_regex

They can be used to configure a list of fact names that will never be sent to PuppetDB, based on exact fact names or regular expressions.

This PR obsoletes: https://github.com/puppetlabs/puppetdb/pull/3998

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/puppetlabs/puppetdb?pullRequest=4021) <br/>All committers have signed the CLA.

djuarezg committed:
Add configuration options to filter facts out in puppetdb termini

This patch works as well for structured facts and is a requirement to overcome
the current limitations of facts-blocklist on PDB configuration options
